src/encoded/audit/experiment.py

```diff
@@ -4627,6 +4622,29 @@ def audit_experiment_mixed_strand_specific_libraries(value, system, excluded_typ
     return
 
 
+def audit_experiment_biosample_mixed_biosamples(value, system, excluded_types):
+    if value['status'] in ['deleted', 'replaced']:
+        return
+
+    biosample = 0
+    pooled = 0
+    for rep in value['replicates']:
+        if 'library' not in rep:
+            continue
+        lib = rep['library']
+        if 'biosample' in lib:
+            biosample += 1
+        elif 'mixed_biosamples' in lib:
+            pooled += 1
+    if biosample > 0 and pooled > 0:
+        detail = (
+            f'This dataset contains libraries with both standard and '
+            f'mixed biosamples.'
+            )
+        yield AuditFailure('inconsistent library structure', detail, level='ERROR')
```

Comment:
I think you should catch this under "inconsistent library biosample", I understand that it is more of "inconsistent libraries in the same experiment", but I don't think we should add a new facet audit entry for this. and "inconsistent library structure" is not self explanantory.

Reply:
I reorganized this as suggested, thanks!